source/faq/questions/using-cypress-faq.md

```diff
@@ -542,3 +541,18 @@ That being said, we use Cypress to test our own Desktop app's front end - by stu
 - Search existing {% url "open issues" https://github.com/cypress-io/cypress/issues %}, it may already be reported!
 - Update Cypress. Your issue may have {% url "already been fixed" changelog %}.
 - {% open_an_issue %}. Your best chance of getting a bug looked at quickly is to provide a repository with a reproducible bug that can be cloned and run.
+
+## {% fa fa-angle-right %} What are your best practices for organizing tests?
+
+We see organizations starting with Cypress by placing end-to-end tests in a separate repo. This is a great practice that allows someone on the team to prototype a few tests and evaluate Cypress without "scaring" everyone else with a new tool. As the time passes and the number of tests grows, we suggest moving end-to-end tests to live right alongside your front-end code. This brings two benefits:
```

Comment:
I'm not sure I agree with this. This is necessary for some projects, but I would outright recommend this. 

I also don't like equating using Cypress with "scaring everyone else" - def does not give a good impression.

Reply:
yeah, scaring is not the word we want to use. But for general test organization, we have to say something. Starting with a separate project for just tests is ok I think, but migrating to have tests closer is much better - developers write tests, etc - same as our dashboard tests.

Reply:
None of our projects have a separate repo for just our tests though, which is why I think it's strange to have that as the first answer to this question.

Reply:
yup, but we do a lot of things differently from a person who is trying to add Cypress to an existing project. Installing it right away on a large project would
- add a huge dependency which is not cached correctly on CI would absolutely cause howls of indignation
- could clash with TypeScript configs for Jest for example
- would immediately start argument: why use Cypress if it does not have cross-browser support

Using it in a separate project at first is the simplest way to try Cypress before jumping head first

Reply:
Maybe you should pull this out of the PR if you want this merged in quickly, cause I think there's general disagreement over this being a 'best practice'.

Reply:
I would actually argue that keeping the tests as close as possible, even modeling the tests to the code as closely as possible is absolutely **essential**.

Best practices are about what's ideal - not workarounds or "starters" for organizations wanting to dip their toes in.

For instance, to get a react application up and running, it's often much "easier and faster" to drop in a `<script src"https://some.cdn/react.js"></script>` right? But that's most certainly not a **best practice**.

Keeping the tests and the code aligned is really the only ideal way to manage a large suite of tests, because you...

1) want to automatically run the tests when the code changes
2) so that when tests fail, you can associate the commit sha which points at the exact snapshot of code
3) and allow developers of the project to continually keep up with and maintain the tests

Pulling the tests out separately would fail all 3 of those things, and brings us back to the antithesis of modern development, which is having QA as a separate department and separate function where the feedback loop is far too long and prone to internal communication breakdowns and errors.

Baking the code, quality and tests directly into the codebase so that engineers have the responsibility of writing and maintaining them is exactly what **shift left** is - and it's the blueprint for modern, healthy, fast, iterative development.

Reply:
This might be good for a "Is Cypress right for me?" section or something emphasizing that writing tests in a separate repo is good for the discovery process/introduction to a large project.

Reply:
I think if it's just a straight up best practice, I'd think that Cypress was a bit pushy thinking I'd be happy to add a new repo to my org just for them.

Reply:
I think we are making a big assumption about people willing to bring big tools and changes to their codebase right away. Starting with tests in a separate repo to show how little is needed to start testing is good practice to start with. Then moving tests to be close to code IS better of course, for all the reasons above, and IS better practice, so whoever wants to phrase it better, be my guest. 

Of course dropping `<script src="react ...>` is NOT the way to production React code. But trying out React like that IS good. And then using create-react-app for example when you know React is for you.

Reply:
Also, why I say "start in a separate repo, then bring it in". When I demo Cypress I don't use local TodoMVC usually, I just point at `todomvc.com`. When I wrote about testing Hacker News for Vue I did not run local fork of the app - instead just point at the url https://vuejsdevelopers.com/2018/01/29/vue-js-e2e-test-hacker-news/ The barrier to trying Cypress should be low, and then a natural progression is to include it with code. The beauty of Cypress is that it does NOT need your code to be successful.

Reply:
I think your approach is valid Gleb - I just don't think its the answer to the question of:

**What are your best practices for organizing tests?**

This isn't a question that says: "How do you recommend first trying out Cypress?" or "How should I go about bringing Cypress into my organization?".

I feel like your answer is fine given that question- but when it comes to **best practices** which is the ideal way in which we believe Cypress should be utilized, suggesting anything short of what we **want** a successful user/organization/adopter of Cypress to look like is a disservice to all of our users.

Reply:
Sorry my comment was out of date - your updated answer is great.